decompose/src/webMain/kotlin/com/arkivanov/decompose/router/stack/webhistory/DefaultWebHistoryController.kt

```diff
@@ -152,22 +154,44 @@ class DefaultWebHistoryController internal constructor(
 
         fun onPopState(state: String?) {
             val newData = state?.let(::deserializeItems) ?: return
-            val newConfiguration = deserializeConfiguration(json = newData.last().configurationJson)
+            val configurations = newData.map {
+                deserializeConfiguration(json = it.configurationJson)
+            }
+
+            doOnPopState(newData, configurations.last(), onWebNavigation(configurations))
```

Comment:
Can we please simplify the logic here? I suggest the following structure, let me know your thoughts.

```kotlin
if (!onWebNavigation(newConfigurations)) {
    // We need to temporary override the listener
    window.setOnPopStateListener { window.setOnPopStateListener(::onPopState) } 
    window.go(stack.value.items.size - newConfigurations.size)
    return
}

// The rest of the existing code goes below
isStateObserverEnabled = false
...
isStateObserverEnabled = true
```